src/axolotl/utils/collators/mm_chat.py

```diff
@@ -19,174 +20,76 @@ class MultiModalChatDataCollator(DataCollatorMixin):
     """
 
     tokenizer: PreTrainedTokenizerBase
-    processor: ProcessorMixin
-    return_tensors: str = "pt"
-    chat_template: Optional[str] = None
+    processing_strategy: ProcessingStrategy
     packing: bool = False
-    max_images: int = -1
+    return_tensors: str = "pt"
     padding: Union[bool, str, PaddingStrategy] = True
     pad_to_multiple_of: Optional[int] = None
 
     def __post_init__(self):
         if self.packing:
             raise ValueError("Packing is currently not supported.")
 
-    def torch_call(
-        self, examples: list[Union[list[int], Any, dict[str, Any]]]
-    ) -> dict[str, Any]:
-        # Handle dict or lists with proper padding and conversion to tensor.
+    def torch_call(self, examples: list[dict]) -> dict[str, Any]:
+        return self.process_rows(examples)
 
-        return self.__class__.process_rows(
-            examples, self.processor, self.chat_template, self.max_images
-        )
+    def process_rows(
+        self,
+        examples: list[dict],
+    ):
+        # Preprocess the examples
+        examples = self.processing_strategy.preprocess(examples)
```

Comment:
should we consider making the ProcessingStrategy class callable (implement `__call__` in place of`preprocess`) ?